Add Cirrus Logic EDB9301 Dev Board to arch/arm/mach-ep93xx

Signed-off-by: H Hartley Sweeten <hsweeten@visionengravers.com>

---

diff --git a/arch/arm/mach-ep93xx/Kconfig b/arch/arm/mach-ep93xx/Kconfig
index 56bddce..9ca8974 100644
--- a/arch/arm/mach-ep93xx/Kconfig
+++ b/arch/arm/mach-ep93xx/Kconfig
@@ -15,6 +15,12 @@ config MACH_ADSSPHERE
 	  Say 'Y' here if you want your kernel to support the ADS
 	  Sphere board.

+config MACH_EDB9301
+	bool "Support Cirrus Logic EDB9301"
+	help
+	  Say 'Y' here if you want your kernel to support the Cirrus
+	  Logic EDB9301 Evaluation Board.
+
 config MACH_EDB9302
 	bool "Support Cirrus Logic EDB9302"
 	help
diff --git a/arch/arm/mach-ep93xx/Makefile
b/arch/arm/mach-ep93xx/Makefile
index 944e42d..7e7536f 100644
--- a/arch/arm/mach-ep93xx/Makefile
+++ b/arch/arm/mach-ep93xx/Makefile
@@ -7,6 +7,7 @@ obj-n			:=
 obj-			:=

 obj-$(CONFIG_MACH_ADSSPHERE)	+= adssphere.o
+obj-$(CONFIG_MACH_EDB9301)	+= edb9301.o
 obj-$(CONFIG_MACH_EDB9302)	+= edb9302.o
 obj-$(CONFIG_MACH_EDB9302A)	+= edb9302a.o
 obj-$(CONFIG_MACH_EDB9307)	+= edb9307.o
diff --git a/arch/arm/mach-ep93xx/edb9301.c
b/arch/arm/mach-ep93xx/edb9301.c
new file mode 100644
index 0000000..4108c7c
--- /dev/null
+++ b/arch/arm/mach-ep93xx/edb9301.c
@@ -0,0 +1,68 @@
+/*
+ * arch/arm/mach-ep93xx/edb9301.c
+ * Cirrus Logic EDB9301 support.
+ *
+ * Copyright (C) 2009 H Hartley Sweeten <hsweeten@visionengravers.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ */
+
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/mm.h>
+#include <linux/sched.h>
+#include <linux/interrupt.h>
+#include <linux/ioport.h>
+#include <linux/mtd/physmap.h>
+#include <linux/platform_device.h>
+#include <linux/io.h>
+#include <linux/i2c.h>
+#include <mach/hardware.h>
+#include <asm/mach-types.h>
+#include <asm/mach/arch.h>
+
+static struct physmap_flash_data edb9301_flash_data = {
+	.width		= 2,
+};
+
+static struct resource edb9301_flash_resource = {
+	.start		= EP93XX_CS6_PHYS_BASE,
+	.end		= EP93XX_CS6_PHYS_BASE + SZ_16M - 1,
+	.flags		= IORESOURCE_MEM,
+};
+
+static struct platform_device edb9301_flash = {
+	.name		= "physmap-flash",
+	.id		= 0,
+	.dev		= {
+		.platform_data	= &edb9301_flash_data,
+	},
+	.num_resources	= 1,
+	.resource	= &edb9301_flash_resource,
+};
+
+static struct ep93xx_eth_data edb9301_eth_data = {
+	.phy_id		= 1,
+};
+
+static void __init edb9301_init_machine(void)
+{
+	ep93xx_init_devices();
+	platform_device_register(&edb9301_flash);
+
+	ep93xx_register_eth(&edb9301_eth_data, 1);
+}
+
+MACHINE_START(EDB9301, "Cirrus Logic EDB9301 Evaluation Board")
+	/* Maintainer: H Hartley Sweeten <hsweeten@visionengravers.com> */
+	.phys_io	= EP93XX_APB_PHYS_BASE,
+	.io_pg_offst	= ((EP93XX_APB_VIRT_BASE) >> 18) & 0xfffc,
+	.boot_params	= EP93XX_SDCE3_PHYS_BASE_SYNC + 0x100,
+	.map_io		= ep93xx_map_io,
+	.init_irq	= ep93xx_init_irq,
+	.timer		= &ep93xx_timer,
+	.init_machine	= edb9301_init_machine,
+MACHINE_END 

